activedirectory.sgnl.yaml

```diff
@@ -0,0 +1,390 @@
+description: "Active Directory as a System of Record"
+hostname: "{{Input Required: Hostname}}"
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "ActiveDirectory-1.0.0"
+# Example Config:
+# {
+#   "baseDN":"dc=org,dc=example,dc=io",
+#   "certificateChain": ". . . base64 encoded certificate chain here . . "
+#   "entityConfig": {
+#     "User": {
+#       "query": "(&(objectCategory=user)(objectClass=user)(distinguishedName=*))"
+#     },
+#     "Computer": {
+#       "query": "(&(objectCategory=computer)(name=*))"
+#     },
+#     "Group": {
+#       "query": "(&(objectCategory=group)(objectClass=group)(distinguishedName=*))"
+#     },
+#     "GroupMember": {
+#       "memberOf": "Group",
+#       "query": "(&(memberOf={{CollectionId}})(objectCategory=user)(objectClass=user))",
+#       "memberUniqueIdAttribute": "memberObjectGUID",
+#       "memberOfUniqueIdAttribute": "groupObjectGUID"
+#     }
+#   },
+# }
+adapterConfig: "ewogICAgImJhc2VETiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBCYXNlRE59fSIsCiAgICAiZW50aXR5Q29uZmlnIjogewogICAgICAgICJVc2VyIjogewogICAgICAgICAgICAicXVlcnkiOiAiKCYob2JqZWN0Q2F0ZWdvcnk9dXNlcikob2JqZWN0Q2xhc3M9dXNlcikoZGlzdGluZ3Vpc2hlZE5hbWU9KikpIgogICAgICAgIH0sCiAgICAgICAgIkNvbXB1dGVyIjogewogICAgICAgICAgICAicXVlcnkiOiAiKCYob2JqZWN0Q2F0ZWdvcnk9Y29tcHV0ZXIpKG5hbWU9KikpIgogICAgICAgIH0sCiAgICAgICAgIkdyb3VwIjogewogICAgICAgICAgICAicXVlcnkiOiAiKCYob2JqZWN0Q2F0ZWdvcnk9Z3JvdXApKG9iamVjdENsYXNzPWdyb3VwKShkaXN0aW5ndWlzaGVkTmFtZT0qKSkiCiAgICAgICAgfSwKICAgICAgICAiR3JvdXBNZW1iZXIiOiB7CiAgICAgICAgICAgICJtZW1iZXJPZiI6ICJHcm91cCIsCiAgICAgICAgICAgICJxdWVyeSI6ICIoJihtZW1iZXJPZj17e0NvbGxlY3Rpb25JZH19KShvYmplY3RDYXRlZ29yeT11c2VyKShvYmplY3RDbGFzcz11c2VyKSkiLAogICAgICAgICAgICAibWVtYmVyVW5pcXVlSWRBdHRyaWJ1dGUiOiAie3tJbnB1dCBSZXF1aXJlZDogSWYgbWVtYmVyT2YgaXMgcHJlc2VudH19IiwKICAgICAgICAgICAgIm1lbWJlck9mVW5pcXVlSWRBdHRyaWJ1dGUiOiAie3tJbnB1dCBSZXF1aXJlZDogSWYgbWVtYmVyT2YgaXMgcHJlc2VudH19IgogICAgICAgIH0KICAgIH0sCiAgICAiY2VydGlmaWNhdGVDaGFpbiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBJZiBMREFQUyBpcyBlbmFibGVkOiBiYXNlNjQgZW5jb2RlZCBjZXJ0aWZpY2F0ZUNoYWlufX0iCn0="
+# All auth mechanisms specified below must be supported by the specified Adapter.
```

Comment:
Let's populate default and valid values for 
- `memberUniqueIdAttribute` and 
- `memberOfUniqueIdAttribute`

The default adapter config in the template should be complete as much as possible to minimize user effort

Reply:
On decoding the default adapterConfig, I see GroupMember configured as
```json
"GroupMember": {
            "memberOf": "Group",
            "query": "(&(memberOf={{CollectionId}})(objectCategory=user)(objectClass=user))",
            "memberUniqueIdAttribute": "memberObjectGUID",
            "memberOfUniqueIdAttribute": "groupObjectGUID"
        }
```

Neither `memberObjectGUID` nor `groupObjectGUID` is present as attributes on  user object fetched via `objectClass=user` and `objectClass=group`. I see `objectGUID` as an attribute on both these entities. What am I missing here?

Reply:
This will be targeted in improvements for adapter, as generic attributes are not currently supported. We'll need the capability in adapter to parse attributes using json path format similar to what we are currently using in other adapters.